File: src/main/java/com/kerimsenturk/labreport/dto/request/CreateReportRequest.java
package com.kerimsenturk.labreport.dto.request;

import com.kerimsenturk.labreport.dto.validator.PatientIdValid;
import com.kerimsenturk.labreport.model.enums.ReportType;

public record CreateReportRequest(
        @PatientIdValid
        String patientId,
        String title,
        String details,
        ReportType reportType ) {
}

File: src/main/web-ui/src/domain/payload/request/CreateReportRequest.js
"use strict";
exports.__esModule = true;
exports.CreateReportRequest = void 0;
var CreateReportRequest = /** @class */ (function () {
    function CreateReportRequest(patientId, title, details, reportType) {
        this.patientId = patientId;
        this.title = title;
        this.details = details;
        this.reportType = reportType;
    }
    return CreateReportRequest;
}());
exports.CreateReportRequest = CreateReportRequest;
